src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolution.cs

```diff
@@ -2914,12 +3039,26 @@ bool hasImplicitConversion(TypeSymbol source, TypeSymbol destination, ref Compou
                 Conversions.ClassifyImplicitConversionFromType(source, destination, ref useSiteInfo).IsImplicit;
         }
 
-        private bool IsBetterParamsCollectionType(TypeSymbol t1, TypeSymbol t2, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
+        private BetterResult IsBetterParamsCollectionType(TypeSymbol t1, TypeSymbol t2, ref CompoundUseSiteInfo<AssemblySymbol> useSiteInfo)
         {
             CollectionExpressionTypeKind kind1 = ConversionsBase.GetCollectionExpressionTypeKind(Compilation, t1, out TypeWithAnnotations elementType1);
             CollectionExpressionTypeKind kind2 = ConversionsBase.GetCollectionExpressionTypeKind(Compilation, t2, out TypeWithAnnotations elementType2);
 
-            return IsBetterCollectionExpressionConversion(t1, kind1, elementType1.Type, t2, kind2, elementType2.Type, ref useSiteInfo);
+            if (kind1 is CollectionExpressionTypeKind.CollectionBuilder or CollectionExpressionTypeKind.ImplementsIEnumerable)
+            {
+                _binder.TryGetCollectionIterationType(CSharpSyntaxTree.Dummy.GetRoot(), t1, out elementType1);
```

Comment:
Note: I could use some ideas on getting a better syntax node here. We don't currently pass a reference to the syntax for the entire method to overload resolution. The syntax node is _mostly_ used for diagnostics, but there is at least one call to see whether calls are omitted in this syntax tree. This _could_ have an impact on binding in particularly interesting cases, where a pattern-based `GetEnumerator` call ends up being seen in a scenario where it should have been omitted, but all that will cause it to do is give us an element type here when it otherwise wouldn't have one, and we likely wouldn't even get here if that were the case, but it's interesting nonetheless.

Reply:
It's surprising we don't have a syntax node representing the containing expression in `MemberResolutionResult` for instance. And to add that, it looks like we'd need to pass the `SyntaxNode` or `BoundExpression` through a number of layers in the call stack for overload resolution.

Reply:
Yup. It's possible to do, but it's quite a bit of work :). I was hoping there'd be something obvious I was missing.

Reply:
I don't think we need to use the syntax node for the call site here, if the only use is determine whether the `params` type is a valid `params` type. That determination is made at the declaration site of the method, and if the type is valid there, it should be valid at call sites where the method is available.